api/v2/app/auth/views.py

```diff
@@ -0,0 +1,71 @@
+import psycopg2
+from datetime import datetime, timedelta
+from flask import Blueprint, jsonify, request, current_app
+from werkzeug.security import check_password_hash, generate_password_hash
+
+
+# local import
+from app.models import User
+from app.auth.helpers import (email_validator,
+                             address_validator,
+                             password_validator,
+                             user_name_validator,
+                             token_required)
+
+auth = Blueprint('auth', __name__)
+
+'''
+Route for Registering new users and throws valid error if the user exists
+'''
+@auth.route('/register', methods=['GET', 'POST'])
+def user_auth():
+    if request.method == 'POST':
+        payload = request.get_json()
+        username = payload['username']
+        email = payload['email']
+        address = payload['address']
+        password = payload['password']
+        conf_pass = payload['confirm_password']
+
+        if username == '' or email == '' or \
+           password == '' or address == '' or conf_pass == '':
+            return jsonify({
+                'message': "Failed you cannot submit empty fields",
+                'status': 'failed'}), 400
```

Comment:
Checking whether fields are empty could be moved into function reused by the endpoints